deploy/tests/test_yaml_api.py

```diff
@@ -0,0 +1,402 @@
+
+@pytest.fixture()
+def run_yamls(scy_hwdb: TmpYaml, scy_runtime: TmpYaml) -> RunTmpYamlSet:
+    return RunTmpYamlSet(scy_hwdb, scy_runtime)
+
+@pytest.fixture()
+def non_existent_file(tmp_path):
+    # tmp_path is builtin pytest fixture to get a per-test temporary directory that should be clean
+    # but we still make sure that it doesn't exist before giving it
+    file = tmp_path / 'GHOST_FILE'
+    file.exists().should.equal(False)
+    return file
+
+@pytest.fixture()
+def firesim_argparser():
+    return firesim.construct_firesim_argparser()
+
+@pytest.fixture()
+def fs_cl2args(firesim_argparser):
+    """fixture that bundles calling parse_args and shlex.split to abbreviate usage in tests"""
+    return lambda x: firesim_argparser.parse_args(shlex.split(x))
+
+@pytest.mark.usefixtures("aws_test_credentials")
+class TestConfigBuildAPI:
+    """ Test config_{build, build_recipes}.yaml APIs """
+
+    # run_test("invalid-build-section")
+    @patch('firesim.buildafi')
+    def test_invalid_build_section(self, buildafi_mock, build_yamls, fs_cl2args):
+        # @patch modifies the symbol table with our Mock but we're dispatching through
+        # a dict and that needs to be patched as well, otherwise our mock won't be called but
+        # the real function will be
+        firesim.BUILD_TASKS['buildafi'] = buildafi_mock
+        os.environ['FIRESIM_SOURCED'] = '1'
+        # at the beginning of the test build_yamls contains the backup-sample-configs
+        # but we can show exactly what we're doing different from the default by
+        build_yamls.build.load(dedent("""
+            builds:
+
+            agfis-to-share:
+                - testing-recipe-name
+
+            share-with-accounts:
+                INVALID_NAME: 123456789012
+            """))
+
+        build_yamls.write()
+        args = fs_cl2args(' '.join(['buildafi']+build_yamls.args))
+        pprint(build_yamls.build.data)
+        firesim.main.when.called_with(args).should.throw(TypeError)
+        buildafi_mock.assert_not_called()
+
+
+    @patch('firesim.buildafi')
+    def test_programmatic_invalid_build_section(self, buildafi_mock, build_yamls, firesim_argparser):
+        # @patch modifies the symbol table with our Mock but we're dispatching through
+        # a dict and that needs to be patched as well, otherwise our mock won't be called but
+        # the real function will be
+        firesim.BUILD_TASKS['buildafi'] = buildafi_mock
+        os.environ['FIRESIM_SOURCED'] = '1'
+        # at the beginning of the test build_yamls contains the backup-sample-configs
+        # but we can show exactly what we're doing different from the default by
+        # by programatically modifying from the sample to remove all the builds
+        #build_yamls.build.data['builds'] = []
+        # lol sometimes, it's hard to tell exactly how the yaml will get loaded because None is what you get
+        build_yamls.build.data['builds'] = None
```

Comment:
Why do we care about the programmatic version of this test? Isn't the main/only API to modify the `builds` section through the file (which is tested in the  `test_invalid_build_section` test?

Reply:
I only did it to show that you can do it this way for testing purposes.  I suppose I should point out that while you're programmatically modifying the YAML, it gets serialized out to a temporary file that gets read in by firesim.

Reply:
Is it necessary for a unique test for this? Aren't the latter tests doing this anyways?

Reply:
No, I'll defiintely remove it before we merge.  Was only trying to have food for discussion in the PR.

Reply:
This is removed now.